python/lsst/ts/m2gui/controltab/tab_diagnostics.py

```diff
@@ -33,7 +44,689 @@ class TabDiagnostics(TabDefault):
         Table's title.
     model : `Model`
         Model class.
+
+    Attributes
+    ----------
+    model : `Model`
+        Model class.
     """
 
     def __init__(self, title, model):
         super().__init__(title, model)
+
+        # Utility data
+        self._power_calibrated = self._create_labels_power()
+        self._power_raw = self._create_labels_power()
+
+        self._force_error_tangent = {
+            "a1": create_label(),
+            "a2": create_label(),
+            "a3": create_label(),
+            "a4": create_label(),
+            "a5": create_label(),
+            "a6": create_label(),
+            "total_weight": create_label(),
+            "tangent_sum": create_label(),
+        }
+
+        self._button_reboot = set_button(
+            "Reboot Controller", self._callback_reboot_controller
+        )
+
+        self._digital_status_input = self._create_indicators_digital_status(
```

Comment:
For me, this makes code extremely difficult to read. I need to find out _create_indicator..to see which indicators will be created. If you drop _get_list_digital_status_input, and put list directly here, you can use len(self._digital_status_input) later instead of len(self._get_list_digital_status_input()), which makes code much easier to read. Long functions isn't a problem here, this is pretty common for GUI buildup functions.

Reply:
For this, I have the different opinion instead. I prefer to avoid the long function in code to improve the readability. This is based on the logic in the "Clean Code: A Handbook of Agile Software Craftsmanship" by Martin Robert C. Thanks!